### DISCUSSION TITLE
Availability reporting - do not log offline when the monitoring is not running (computer off, Chrome closed, etc)

### DISCUSSION THREAD

**Discussion by rqueizan** [InternetConnectionMonitor/icm-public#104] [Ideas and Feature requests]
I'm going to put it in context:
If at 11 PM I turn off the computer, at which time the Internet is not available. I turn it on in the morning at 8 AM with internet available. In the statistics it reports me without Internet from 12 AM to 8 AM. Reporting a very low percentage of availability, although I have just turned on the computer.

If, on the contrary, at 11PM I turn off the computer while I had the Internet available. I turn it on at 8 AM with internet available. In the statistics it reports 100% availability, which is closer to reality.

I believe that on a day-to-day basis you should take the first Internet connection as a point of reference, and not assume that it was previously available or not, which I think is what is happening. Or if you're going to assume it, take it as available by default.

![image](https://github.com/user-attachments/assets/268c0850-467c-4a69-8ed2-27ecd03d65f3)

![image](https://github.com/user-attachments/assets/876cf1f0-cb50-40e3-9c46-fd771ac54bb3)

**Duplicated reports for this issue (some contain more context):**
- #100 
- #123

**Comment by dmishh** [comment]
@rqueizan thanks for the report! This is how the extension works now and how it was intended to work, so I'm converting this to a feature request. It would require some additional logic to understand if the computer is off VS internet is down. I'll think about it

**Comment by dmishh** [comment]
Related to #100 - which is the same idea, but for On/Off ICM state

**Reply by rqueizan** [reply]
My idea of the situation would be to take the first reading of the day as a starting point for the rest of the calculations.
If at that moment of time there is no Internet, start counting the time without Internet service. If, on the other hand, there is Internet, start counting time with Internet service.
This would prevent miscalculation and ensure more accuracy in the data.

**Comment by dmishh** [comment]
There should be an extra status "No data" which will indicate that ICM is not monitoring the status and these periods should be excluded from the availability calculation

<img width="634" alt="image" src="https://github.com/user-attachments/assets/8da74f45-24e7-452d-b502-420a21b36424">


To test: when computer is in sleep, there may be a lot of missing data periods

**Reply by Zulkiers** [reply]
Exactly what I was thinking!

**Reply by Zulkiers** [reply]
Is this in the works?

**Comment by dmishh** [comment]
Will be implemented in #157
